scripts/include/helpers.inc.php

```diff
@@ -121,45 +121,97 @@ function mergeXliffDomToAtom(DOMDocument $domA, DOMDocument $domB): DOMDocument
 
 function mergeXliffDomToWeblate(DOMDocument $domA, DOMDocument $domB): DOMDocument
 {
-  // Elements in A but not in B -> copy to B
-  $sourcesA = $domA->getElementsByTagName('source');
-  $sourcesB = $domB->getElementsByTagName('source');
+  // Elements in A AND B -> check source string and if different -> replace in B
+  $transUnitsA = $domA->getElementsByTagName('trans-unit');
+  $transUnitsB = $domB->getElementsByTagName('trans-unit');
 
-  foreach ($sourcesA as $source) {
-    $stringsA[] = $source->nodeValue;
+  $i = 0;
+  foreach ($transUnitsA as $transUnit) {
```

Comment:
`$transUnit` isn't used in this loop, so I would suggest either making it "for" loop (i.e. `for $i = 0; i < len($transUnitsA); i++ {`) or assigning `$transUnit` by reference and using it in the loop (i.e. `foreach ($transUnitsA as &$transUnit) {`).  As it is now, the value of `transUnitsA` is being copied to `$transUnit` unnecessarily.

Reply:
Good idea - I like this better. Updating....